api/server.js

```diff
@@ -105,54 +186,104 @@ app.post("/bookshelves/remove", async (req, res) => {
  * If no books match search criteria... return an empty JSON Object
  */
 app.get("/books/search/:title/:author/:genre/:offset/:resultLength", async function(req, res) {
-  try {
-    const db = await getDBConnection();
-    await db.close();
-  } catch (err) {
-    loggingModule(err, "bookSearch");
-  }
+	try {
+		
+	} catch (err) {
+		loggingModule(err, "bookSearch");
+	}
 });
 
 /**
  * Gets detailed information for book based on ISBN
  */
 app.get("/books/detail/:isbn", async function(req, res) {
-  try {
-    const db = await getDBConnection();
-    await db.close();
-  } catch (err) {
-    loggingModule(err, "bookDetail");
-  }
+	try {
+		
+	} catch (err) {
+		loggingModule(err, "bookDetail");
+	}
 });
 
 /* -----------------  HELPER FUNCTIONS  ---------------- */
-async function getDBConnection() {
-  const db = mysql.createPool({
-    host: process.env.DB_URL || 'localhost',
-    port: process.env.DB_PORT || '8889',
-    user: process.env.DB_USERNAME || 'root',
-    password: process.env.DB_PASSWORD || 'root',
-    database: process.env.DB_NAME || DB_NAME
-  });
-  return db;
+
+
+/**
+ * Creates new User based on info
+ * @param {String[]} info
+ */
+ async function createUser(info) {
+	let query = "INSERT INTO User (username, password) VALUES (?, ?);";
+	await db.query(query, info);
+}
+/**
+ * Updates the User's current Color Scheme. 
+ * @param {String[]} info 
+ */ 
+async function updateColorScheme(info) {
+	let query = "UPDATE User SET color_scheme = ? WHERE username = ?";
+	await db.query(query, info);
+}
+
+/**
+ * Gets password from username
+ * @param {String} username
+ * @returns info of username
+ */
+async function getPassword(username) {
+	let query = "SELECT * FROM User WHERE username=?;";
+	let [rows] = await db.query(query, [username]);
+	return rows;
+}
+
+/**
+ * Gets the book from the specified bookshelf
+ * @param {String[]} info
+ * @returns the Bookshelf information of the user.
+ */
+async function geBook(info) {
+	let query = "SELECT * FROM User WHERE username = ?;"; // Unsure how to address the other parts of the query.
+	let [rows] = await db.query(query, [username]);
+	return rows;
+}
+
+/**
+ * Checks if username exists
+ * @param {String} username
+ * @returns {boolean} true if username already exists
+ */
+async function checkIfExist(username) {
+	let query = "SELECT * FROM User WHERE username = ?;";
+	let [rows] = await db.query(query, [username]);
+	return (rows.length >= 1);
+}
+
+/**
+ * Checks if the given color scheme exists. 
+ * @param {String} color_scheme
+ * @returns {boolean} true if the given color scheme exists
+ */
+async function checkColor(color_scheme) {
+	let query = "UPDATE User SET color_scheme = ? WHERE username = ?";
```

Comment:
This SQL statement updates the database, it doesn't check for an existing color scheme. We should use an `SELECT` SQL statement instead.